spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/neo4j/Neo4jHealthIndicator.java

```diff
@@ -52,12 +54,28 @@ public Neo4jHealthIndicator(SessionFactory sessionFactory) {
 		this.sessionFactory = sessionFactory;
 	}
 
+	/**
+	 * Create a new {@link Neo4jHealthIndicator} using the specified
+	 * {@link SessionFactory} and a custom Cypher validation query.
+	 * @param sessionFactory the SessionFactory
+	 * @param cypher the Cypher validation statement
+	 */
+	public Neo4jHealthIndicator(SessionFactory sessionFactory, String cypher) {
+		this.sessionFactory = sessionFactory;
+		CYPHER = cypher;
+	}
+
 	@Override
 	protected void doHealthCheck(Health.Builder builder) throws Exception {
 		Session session = this.sessionFactory.openSession();
 		Result result = session.query(CYPHER, Collections.emptyMap());
-		builder.up().withDetail("nodes",
-				result.queryResults().iterator().next().get("nodes"));
+		Map<String, Object> details = new HashMap<>();
+
+		for (Map<String, Object> queryResult : result.queryResults()) {
+			details.putAll(queryResult);
+		}
+
+		builder.up().withDetails(details);
```

Comment:
This should go away IMO, that's unrelated to making the cyper configurable.

Reply:
The previous code relies on a "nodes" value to be available in the results returned from Neo4j. As this PR makes the Cypher validation statement configurable, the code can no longer rely on this value to be present. 

The proposed code change instead iterates over the returned results, putting them into a Map, and stores this Map as the details. This way whatever results are present will be made available as details.

Do you have suggestions on how this can be improved?

Reply:
I am afraid I don't know much Neo4j but adding "whatever results are present" doesn't sound good to me. There must be some sort of minimal set of data (we can document what the cypher can look like).

Regardless, there is absolutely no reason to change the API of the builder for that.